### PR TITLE
Fix #2097: LocalDate.range returns incorrect result

### LINKED ISSUE #2097: LocalDate.range returns incorrect result
Given an instance `d` of `LocalDate`, `d.range(ChronoField.DAY_OF_MONTH)` returns a range of 1 - 28/31, when on the JVM the upper bound is adapted to the length of the month of `d`. For instance, `LocalDate.of(2012, 2, 1).range(ChronoField.DAY_OF_MONTH)` should return a range of 1 - 29.

### PR DESCRIPTION
**Pull request by ummels**

### CODE REVIEW

File: test-suite/js/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/temporal/TemporalAccessorTest.scala [hunk 1]
```diff
@@ -17,7 +17,7 @@ trait TemporalAccessorTest extends JasmineTest with ExpectExceptions {
     it("should respond to `range`") {
       for {
         accessor <- samples
-        field <- ChronoField.values()
+        field <- ChronoField.values() if field != ChronoField.DAY_OF_MONTH
```
Comment (ummels) [inline test-suite/js/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/temporal/TemporalAccessorTest.scala]: Just a hack to make the test pass. The correct test is included in #2096.

Comment (nicolasstucki) [inline test-suite/js/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/temporal/TemporalAccessorTest.scala]: Put the correct test directly in this commit.

Comment (ummels) [inline test-suite/js/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/temporal/TemporalAccessorTest.scala]: Fixed. Would be better to have the test in `LocalDateTest.scala`, but we need to add the exception here anyway and it's only temporarily anyway.

**Comment by nicolasstucki** [comment]
Jenkins, test this please.

**Comment by nicolasstucki** [comment]
LGTM

**Pull request merged**